src/core/sorted_map.cc

```diff
@@ -770,10 +770,7 @@ SortedMap* SortedMap::FromListPack(PMR_NS::memory_resource* res, const uint8_t*
 }
 
 bool SortedMap::DefragIfNeeded(float ratio) {
-  auto cb = [this](sds old_obj, sds new_obj) {
-    score_tree->Delete(old_obj);
-    score_tree->Insert(new_obj);
-  };
+  auto cb = [this](sds old_obj, sds new_obj) { score_tree->ForceUpdate(old_obj, new_obj); };
```

Comment:
when I checked if `old_obj` is still alive when thsi callback is called, I realised that `ReallocIfNeededGeneric` is really  a bad name because it does not REALLOCATE.  The better name would be `DuplicateEntryIfFragmented`